dojo/tools/whitehat_sentinel/parser.py

```diff
@@ -0,0 +1,238 @@
+
+import json
+import logging
+import re
+from json import JSONDecodeError
+from typing import Union, List
+from urllib.parse import urlparse
+
+from dojo.models import Finding, Endpoint
+
+
+class WhiteHatSentinelParser(object):
+    """
+    A class to parse WhiteHat Sentinel vulns from the WhiteHat Sentinel API vuln?query_site=[
+    SITE_ID]&format=json&display_attack_vectors=all&display_custom_risk=1&display_risk=1&display_description=custom
+    """
+
+    _LOGGER = logging.getLogger(__name__)
+
+    def get_scan_types(self):
+        return ["WhiteHat Sentinel"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "WhiteHat Sentinel"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "WhiteHat Sentinel output from api/vuln/query_site can be imported in JSON format."
+
+    def get_findings(self, file, test):
+
+        # Exit if file is not provided
+        if file is None:
+            return []
+
+        try:
+            findings_collection = json.load(file)
+        except JSONDecodeError:
+            return []
+
+        if not findings_collection.keys():
+            return list()
+
+        # Make sure the findings key exists in the dictionary and that it is not null or an empty list
+        if 'collection' not in findings_collection.keys() or not findings_collection['collection']:
+            return list()
```

Comment:
Better launch ValueError
```suggestion
        # Make sure the findings key exists in the dictionary and that it is not null or an empty list
        if 'collection' not in findings_collection.keys() or not findings_collection['collection']:
            raise ValueError('collection key not present in the file. The report seems to be not well-formed.')
```